aptos_sdk/async_client.py

```diff
@@ -573,16 +572,14 @@ async def wait_for_transaction(self, txn_hash: str) -> None:
         Waits up to the duration specified in client_config for a transaction to move past pending
         state.
         """
-
-        count = 0
-        while await self.transaction_pending(txn_hash):
-            assert (
-                count < self.client_config.transaction_wait_in_seconds
-            ), f"transaction {txn_hash} timed out"
-            await asyncio.sleep(1)
-            count += 1
-
-        response = await self._get(endpoint=f"transactions/by_hash/{txn_hash}")
+        while True:
```

Comment:
Let's think through this a bit more:
1. We want to limit the maximum time a transaction can be waited on -- that feature is lost in this code
2. `wait_by_hash` has a maximum waiting period, so we can still get an approximation here
3. What if wait time is 0, then we probably want to use the legacy method
4. If we get a 404, we're just going to spin on querying the end point, we probably should sleep, also should document the internal behavior

Can you think of a different pattern that would satisfy the requirements?

Reply:
Can you review again for my first commit https://github.com/aptos-labs/aptos-python-sdk/commit/c3003a9f4ec4457a933862b4e6b291d8f7d875af? 😂

I think that commit should satisfy 1,  2 and 4. For 3, adjust 
```python
await asyncio.wait_for(
    self._wait_for_txn(txn_hash),
    timeout=self.client_config.transaction_wait_in_seconds,
)
```
to
```python
await asyncio.wait_for(
    self._wait_for_txn(txn_hash),
    timeout=self.client_config.transaction_wait_in_seconds if self.client_config.transaction_wait_in_seconds else None,
)
```

What do you think?